domain/src/main/java/org/oppia/android/domain/topic/TopicListController.kt

```diff
@@ -794,51 +839,88 @@ class TopicListController @Inject constructor(
   }
 
   // TODO(#5344): Remove this in favor of per-classroom data handling.
-  private fun loadClassroom(): ClassroomRecord {
+  private fun loadClassrooms(): List<ClassroomRecord> {
     return if (loadLessonProtosFromAssets) {
-      return assetRepository.loadProtoFromLocalAssets(
+      assetRepository.loadProtoFromLocalAssets(
         assetName = "classrooms",
-        baseMessage = ClassroomList.getDefaultInstance()
-      ).classroomsList.single() // Only one record is currently expected.
-    } else loadClassroomFromJson()
+        baseMessage = ClassroomIdList.getDefaultInstance()
+      ).classroomIdsList.map {
+        assetRepository.loadProtoFromLocalAssets(
+          assetName = it,
+          baseMessage = ClassroomRecord.getDefaultInstance()
+        )
+      }
+    } else loadClassroomsFromJson()
   }
 
   // TODO(#5344): Remove this in favor of per-classroom data handling.
-  private fun loadClassroomFromJson(): ClassroomRecord {
-    val classroomsObj = jsonAssetRetriever.loadJsonFromAsset("classrooms.json")
-    checkNotNull(classroomsObj) { "Failed to load classrooms.json." }
-    val classroomArray = classroomsObj.optJSONArray("classrooms")
-    checkNotNull(classroomArray) { "classrooms.json missing classrooms array." }
-    check(classroomArray.length() == 1) { "Expected classrooms.json to have one single classroom." }
-    val classroom = checkNotNull(classroomArray.optJSONObject(0)) { "Expected non-null classroom." }
-    val topicPrereqsObj = checkNotNull(classroom.optJSONObject("topic_prerequisites")) {
-      "Expected classroom to have non-null topic_prerequisites."
-    }
-    val topicPrereqs = topicPrereqsObj.keys().asSequence().associateWith { topicId ->
-      val topicIdArray = checkNotNull(topicPrereqsObj.optJSONArray(topicId)) {
-        "Expected topic $topicId to have a non-null string list."
+  private fun loadClassroomsFromJson(): List<ClassroomRecord> {
+    // Load the classrooms.json file.
+    val classroomIdsObj = jsonAssetRetriever.loadJsonFromAsset("classrooms.json")
+    checkNotNull(classroomIdsObj) { "Failed to load classrooms.json." }
+    val classroomIds = classroomIdsObj.optJSONArray("classroom_id_list")
+    checkNotNull(classroomIds) { "classrooms.json missing classroom IDs." }
+
+    // Initialize a list to store the classroom records.
+    val classroomRecords = mutableListOf<ClassroomRecord>()
+
+    // Iterate over all classroom IDs and load each classroom's JSON.
+    for (i in 0 until classroomIds.length()) {
+      val classroomId = checkNotNull(classroomIds.optString(i)) {
+        "Expected non-null classroom ID at index $i."
       }
```

Comment:
Can potentially be more idiomatic.
```suggestion
 classroomIds.forEachIndexed { index, _ ->
    val classroomId = checkNotNull(classroomIds.optString(index)) {
        "Expected non-null classroom ID at index $index."
    }
```

Reply:
This cannot be done, since `classroomIds` is a `JSONArray` here whereas `forEachIndexed` can be performed only on a Kotlin Array.

Reply:
Okay, sounds good.